### PR TITLE
Update uci.cpp

### PR DESCRIPTION
**Pull request by Amansingh0807**
enhanced the UCI loop code for the Sonic chess engine by improving thread safety and error handling. I added validation for user commands and clearer error messages, especially in the setoption and parse_position functions. Additionally, I recommended replacing magic numbers with named constants for better readability and suggested including available command options in the Unknown Command response to improve user experience.

### CODE REVIEW

**Review by fffelix-huang — CHANGES_REQUESTED** [review]
Hi,

Thanks for the PR! I've reviewed your code, and there are some changes that should be made. One suggestion is to ensure consistent spacing after `if`, `while`, and `for` statements throughout the codebase. Would you be interested in adding a `.clang-format` configuration and a workflow test?

**Comment by Amansingh0807** [comment]
> Hi,
> 
> Thanks for the PR! I've reviewed your code, and there are some changes that should be made. One suggestion is to ensure consistent spacing after `if`, `while`, and `for` statements throughout the codebase. Would you be interested in adding a `.clang-format` configuration and a workflow test?

sure

File: src/uci.cpp [hunk 1]
```diff
@@ -17,9 +17,13 @@
 #include "search.h"
 #include "ucioption.h"
 #include "version.h"
+#include <mutex> // Added for thread safety
```
[the remark this answers is not in the document]
Comment (Amansingh0807) [reply src/uci.cpp:20]: ok doing it

File: src/uci.cpp [hunk 2]
```diff
@@ -37,143 +41,185 @@ void uci_loop() {
     SearchInfo search_info;
     std::string cmd;
     std::thread th;
-    while(std::getline(std::cin, cmd)) {
-        if(cmd == "") {
+
+    while (std::getline(std::cin, cmd)) {
+        if (cmd.empty()) {
```
[the remark this answers is not in the document]
Comment (Amansingh0807) [reply src/uci.cpp:46]: i think Changing while loop with empty command checks for better input handling if you want to do as it was earlier i will change it accordingly.

[the remark this answers is not in the document]
Comment (fffelix-huang) [reply src/uci.cpp:49]: Yeah what I mean yours is better. I'm not asking changes on this line 😊

[the remark this answers is not in the document]
Comment (Amansingh0807) [reply src/uci.cpp:49]: ohh sorry I couldn't Recognize it.

**Pull request merged**